src/sail_sv_backend/jib_sv.ml

```diff
@@ -592,17 +604,22 @@ module Make (Config : CONFIG) = struct
     in
     Str.string_match regexp s 0
 
-  let pp_id_string id =
+  let is_sv_global s =
+    match !sv_globals with
+    | None -> failwith "USED is_sv_global BUT sv_globals IS UNSET"
+    | Some smap -> StringSet.mem s smap
+
+  let pp_id_string ?(prepend_globals = true) id =
     let s = string_of_id id in
     if
       valid_sv_identifier s
       && (not (has_bad_prefix s))
       && (not (StringSet.mem s Keywords.sv_reserved_words))
       && not (StringSet.mem s Keywords.sv_used_words)
-    then s
+    then if Config.no_toplevel_globals && is_sv_global s && prepend_globals then "`SAIL_GLOBALS." ^ s else s
```

Comment:
Rather than using the `prepend_globals` flag and `is_sv_global` this function should take `spec_info` as an argument to get the information placed there (in line 553 of this PR).